src/sentry/incidents/models/incident.py

```diff
@@ -60,28 +60,39 @@ class Meta:
 
 class IncidentManager(BaseManager["Incident"]):
     CACHE_KEY = "incidents:active:%s:%s"
+    SUB_CACHE_KEY = "incidents:active:%s:%s:%s"
 
     def fetch_for_organization(self, organization, projects):
         return self.filter(organization=organization, projects__in=projects).distinct()
 
     @classmethod
-    def _build_active_incident_cache_key(cls, alert_rule_id, project_id):
+    def _build_active_incident_cache_key(cls, alert_rule_id, project_id, subscription_id):
+        if subscription_id:
+            return cls.SUB_CACHE_KEY % (alert_rule_id, project_id, subscription_id)
         return cls.CACHE_KEY % (alert_rule_id, project_id)
 
-    def get_active_incident(self, alert_rule, project):
-        cache_key = self._build_active_incident_cache_key(alert_rule.id, project.id)
+    def get_active_incident(self, alert_rule, project, subscription=None):
+        """
+        fetches the latest incident for a given alert rule and project (and subscription) that is not closed
+        """
+        cache_key = self._build_active_incident_cache_key(
+            alert_rule_id=alert_rule.id,
+            project_id=project.id,
+            subscription_id=(subscription.id if subscription else None),
+        )
         incident = cache.get(cache_key)
         if incident is None:
             try:
-                incident = (
-                    Incident.objects.filter(
-                        type=IncidentType.ALERT_TRIGGERED.value,
-                        alert_rule=alert_rule,
-                        projects=project,
-                    )
-                    .exclude(status=IncidentStatus.CLOSED.value)
-                    .order_by("-date_added")[0]
+                incident_query = Incident.objects.filter(
+                    type=IncidentType.ALERT_TRIGGERED.value,
+                    alert_rule=alert_rule,
+                    projects=project,
                 )
+                if subscription:
+                    incident_query = incident_query.filter(subscription=subscription)
```

Comment:
If we don't have a subscription, should we filter to rows with a null subscription value?

Reply:
Ah - that's a good call 🤔